gym_bridges/envs/bridges_env.py

```diff
@@ -261,11 +268,96 @@ def reset(self, state=None, gap_count=None):
 
         return self._state.copy()
 
+    def draw_state(
+        self, screen: pygame.Surface, textures: dict[StateType, pygame.Surface]
+    ):
+        x_coordinates = np.arange(0, self.shape[1])
+        y_coordinates = np.arange(0, self.shape[0])
+
+        block_size = renderer_config.BLOCK_SIZE * 0.5
+
+        x_window_coordinates = np.arange(
+            renderer_config.WINDOW_WIDTH // 2 - block_size * self.shape[1] // 2,
+            renderer_config.WINDOW_WIDTH // 2 + block_size * self.shape[1] // 2,
+            block_size,
+        )
+        y_window_coordinates = np.arange(
+            renderer_config.WINDOW_HEIGHT // 2 - block_size * self.shape[0] // 2,
+            renderer_config.WINDOW_HEIGHT // 2 + block_size * self.shape[0] // 2,
+            block_size,
+        )
+
+        for x_coordinate, x_window_coordinate in zip(
+            x_coordinates, x_window_coordinates
+        ):
+            for y_coordinate, y_window_coordinate in zip(
+                y_coordinates, y_window_coordinates
+            ):
+                rect = pygame.Rect(
+                    x_window_coordinate,
+                    y_window_coordinate,
+                    block_size,
+                    block_size,
+                )
+                if (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.GROUND
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.GROUND], rect)
+                elif (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.BRICK
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.BRICK], rect)
+                elif (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.EMPTY
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.EMPTY], rect)
+
     def render(self, mode="human"):
-        mapping = {
-            BridgesEnv.StateType.GROUND: "@@",
-            BridgesEnv.StateType.BRICK: "[]",
-            BridgesEnv.StateType.EMPTY: "  ",
-        }
-        flat_repr = tuple([mapping[x] for x in self._state.flatten()])
-        print((("%s" * self.shape[1] + "\n") * self.shape[0]) % flat_repr)
+        if mode == "human":
+            mapping = {
+                BridgesEnv.StateType.GROUND: "@@",
+                BridgesEnv.StateType.BRICK: "[]",
+                BridgesEnv.StateType.EMPTY: "  ",
+            }
+            flat_repr = tuple([mapping[x] for x in self._state.flatten()])
+            print((("%s" * self.shape[1] + "\n") * self.shape[0]) % flat_repr)
+        elif mode == "pygame":
+            if not self._pygame_initialized:
```

Comment:
prefer to move this logic into a private function like 

def _initialize_pygame_if_necessary() -> None:
  if self._pygame_initialized:
     return
  // the rest of the code here to do the init work.

That makes this render() function shorter.

Reply:
Per discussion, I think it's fine for the self._screen just to get initialized in the _if_necessary function.